contracts/modules/ModuleBaseWithFee.sol

```diff
@@ -125,6 +149,50 @@ contract ModuleBaseWithFee is ModuleBase {
         _transfer(_token, _to, amountAfterFee);
     }
 
+    /**
+     * @notice              Transfers a token amount with automated fee payment
+     * @param _token        Token in which the transfer happens
+     * @param _rewardAmount Amount of the transfer
+     * @param _daoplomats    Array containing the DAOplomat address that will receive the
+                                DAOplomat reward.
+     * @param _rewardPathTo  Array containing the amount of reward each DAOplomat receives.
+     */
+    function _payDaoplomatReward(
+        address _token,
+        uint256 _rewardAmount,
+        address[] memory _daoplomats,
+        uint256[] memory _rewardPathTo
+    ) internal {
+        for (uint256 i; i < _daoplomats.length; i++) {
+            uint256 reward = (_rewardAmount * _rewardPathTo[i]) / BPS;
+            _transfer(_token, _daoplomats[i], reward);
+        }
+    }
+
+    /**
+     * @notice                          Transfers a token amount with automated fee payment
+     * @param _token                    Token in which the transfer happens
+     * @param _to                       Target of the transfer
+     * @param _amount                   Amount of the transfer
+     * @param _percentDaoplomatReward   Total percentage of DAOplomat reward in basepoints
+     * @return amountDaoplomatReward    The amount minus the fee
+     */
+    function _transferWithFeeAndReturnReward(
+        address _token,
+        address _to,
+        uint256 _amount,
+        uint256 _percentDaoplomatReward
+    ) internal returns (uint256 amountDaoplomatReward) {
+        amountDaoplomatReward = (_amount * _percentDaoplomatReward) / BPS;
+
+        uint256 amountAfterFee = _payFeeAndReturnRemainder(_token, _amount);
+        _transfer(
+            _token,
+            _to,
+            _amount - (amountAfterFee - amountDaoplomatReward)
```

Comment:
Are you sure this works?

Assume these values:
* Amount = `10`
* Fee = `2`
* Reward = `1`
* Amount After Fee = `10 - 2` = `8`

Then this line calculates:
`10 - (8 - 1) = 10 - 7 = 3`

So only `3`, which is the fee and the reward are being transferred. This doesn't look right to me, or am I missing something?

Reply:
This is indeed wrong. Changed it